ingestion/src/metadata/utils/streamable_logger.py

```diff
@@ -0,0 +1,515 @@
+                return
+
+            # Format the log record
+            log_entry = self.format(record)
+
+            # Try to add to queue (non-blocking)
+            try:
+                self.log_queue.put_nowait(log_entry)
+            except queue.Full:
+                # Queue is full, fallback to local logging
+                logger.warning("Log queue is full, falling back to local logging")
+                self.fallback_handler.emit(record)
+
+        except Exception as e:
+            # Any error, fallback to local logging
+            logger.error(f"Error in emit: {e}")
+            try:
+                self.fallback_handler.emit(record)
+            except Exception:
+                pass  # Last resort: silently drop the log
+
+    def flush(self):
+        """Flush any buffered logs"""
+        # Signal worker to flush by adding a None marker
+        try:
+            self.log_queue.put_nowait(None)
+        except queue.Full:
+            pass
+
+    def get_metrics(self) -> Dict[str, Any]:
+        """Get current metrics for monitoring"""
+        return {
+            **self.metrics,
+            "circuit_state": self.circuit_breaker.state.value,
+            "queue_size": self.log_queue.qsize(),
+            "worker_alive": self.worker_thread.is_alive()
+            if self.worker_thread
+            else False,
+        }
+
+    def close(self):
+        """Close the handler and cleanup resources"""
+        if self.enable_streaming and self.worker_thread:
+            # Log final metrics
+            logger.info(f"StreamableLogHandler metrics: {self.get_metrics()}")
+
+            # Signal worker to stop
+            self.stop_event.set()
+
+            # Wait for worker to finish (with timeout)
+            if self.worker_thread.is_alive():
+                self.worker_thread.join(timeout=5.0)
+
+            # Close the session
+            self.session.close()
+
+        # Close fallback handler
+        self.fallback_handler.close()
+
+        super().close()
+
+
+# Global handler instance for cleanup
+_streamable_handler: Optional[StreamableLogHandler] = None
+
+
+def setup_streamable_logging_for_workflow(
+    metadata: OpenMetadata,
+    pipeline_fqn: Optional[str] = None,
+    run_id: Optional[UUID] = None,
+    log_level: int = logging.INFO,
+) -> Optional[StreamableLogHandler]:
+    """
+    Setup streamable logging for a workflow execution.
+    This is automatically called when a workflow starts if:
+    1. The server has log storage configured
+    2. The pipeline FQN and run ID are available
+    3. ENABLE_STREAMABLE_LOGS environment variable is set to 'true' (optional)
+
+    Args:
+        metadata: OpenMetadata client instance
+        pipeline_fqn: Fully qualified name of the pipeline
+        run_id: Unique run identifier
+        log_level: Logging level
+
+    Returns:
+        StreamableLogHandler instance if configured, None otherwise
+    """
+    global _streamable_handler
```

Comment:
since we have already a centralized logger we use everywhere, we should tie this logic in it